### PR TITLE
Add VUID 01389

### PR DESCRIPTION
**Pull request by sfricke-samsung**
VUID-VkMappedMemoryRange-size-01389

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -10755,22 +10755,30 @@ bool CoreChecks::ValidateMappedMemoryRangeDeviceLimits(const char *func_name, ui
...
+            const VkDeviceSize allocationSize = mem_info->alloc_info.allocationSize;
+            if ((size != VK_WHOLE_SIZE) && (size + offset != allocationSize) && (SafeModulo(size, atom_size) != 0)) {
                 skip |= LogError(mem_ranges->memory, "VUID-VkMappedMemoryRange-size-01390",
                                  "%s: Size in pMemRanges[%d] is 0x%" PRIxLEAST64
                                  ", which is not a multiple of VkPhysicalDeviceLimits::nonCoherentAtomSize (0x%" PRIxLEAST64 ").",
-                                 func_name, i, mem_ranges[i].size, atom_size);
+                                 func_name, i, size, atom_size);
+            } else if ((size == VK_WHOLE_SIZE) && SafeModulo(allocationSize - offset, atom_size) != 0) {
+                skip |= LogError(mem_ranges->memory, "VUID-VkMappedMemoryRange-size-01389",
+                                 "%s: Size in pMemRanges[%d] is VK_WHOLE_SIZE and allocationSize miuns offset (0x%" PRIxLEAST64
```
Comment (ncesario-lunarg) [inline layers/core_validation.cpp:10778]: `miuns` -> `minus`?

Comment (sfricke-samsung) [reply layers/core_validation.cpp:10778]: 🤦 will batch up fixes

**Review by ncesario-lunarg — APPROVED** [review]
LGTM and is passing internal CI 👍

**Comment by ncesario-lunarg** [comment]
Running update through internal CI.

**Comment by ncesario-lunarg** [comment]
Passing internal CI and LGTM 👍

**Pull request merged**